x/curating/keeper/upvote.go

```diff
@@ -44,16 +36,30 @@ func (k Keeper) CreateUpvote(
 	}
 
 	if !found {
-		// no deposit is locked
 		// this curator gets both creator + curator rewards (sent to reward_account)
 		err = k.CreatePost(ctx, vendorID, postID, "", nil, rewardAccount)
 		if err != nil {
 			return err
 		}
 	}
 
-	voteAmt := k.voteAmount(ctx, int64(voteNum))
-	upvote := types.NewUpvote(curator, rewardAccount, voteAmt, ctx.BlockTime())
+	// check if there is already an upvote, and append vote num
+	upvote, found, err := k.GetUpvote(ctx, vendorID, postID, curator)
+	if err != nil {
+		return err
+	}
+	if found {
+		voteNumNew := voteNum + upvote.VoteNum
+		voteAmtNew := k.voteAmount(ctx, int64(voteNumNew))
```

Comment:
I think we will need to update the amount of coins send to the pool if the upvote is updated, or it will be overpaying for the upvote
voteAmt := newVoteAmt - previousAmt

```err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, curator, types.VotingPoolName, sdk.NewCoins(voteAmt))```

Reply:
Thanks! Check now.